### ISSUE TITLE
k_delayed_work_cancel() failed to return -EINVAL

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1250]
**_Reported by Sharron LIU:_**

Cancel a delayed work item after it has completed its work, k_delayed_work_cancel() failed to return -EINVAL as stated in the "kernel.h"

To help quick locate the issue, modify the legacy "test_nano_work", by adding the testpoint. See attache patch to the test case.

```
# source zephyr-env.sh
# make BOARD=qemu_x86 qemu
# verify "FAIL - test_delayed_cancel@256. return from cancel: 0, expected -EINVAL"
```

(Imported from Jira ZEP-1373)

**Comment by nashif** [comment]
**_by Sharron LIU:_**

* [test_delayed_work_cancel.patch](https://s3.amazonaws.com/logs.zephyrproject.org/jira/ZEP-1373/test_delayed_work_cancel.patch) (size: 882)

**Comment by nashif** [comment]
**_by Benjamin Walsh:_**

* [test_delayed_work_cancel-ben.patch](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-1250/test_delayed_work_cancel-ben.patch) (size: 1671)

**Comment by nashif** [comment]
**_by Benjamin Walsh:_**

_Sharron LIU_  The test does not pass anymore with the fix applied. The patch should be [^test_delayed_work_cancel-ben.patch] instead.

**Comment by nashif** [comment]
**_by Sharron LIU:_**

Verified the patch works. Thanks.

**Comment by nashif** [comment]
**_by Sharron LIU:_**

Verified issue fixed in v1.6
Closing this. Thanks.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Fixing incorrect priority
